src/tiledb/cloud/sql/tiledb_cursor.py

```diff
@@ -0,0 +1,117 @@
+import pyarrow as pa
+
+from tiledb.cloud.sql._execution import exec
+from tiledb.cloud.sql.db_api_exceptions import DataError
+from tiledb.cloud.sql.db_api_exceptions import NotSupportedError
+from tiledb.cloud.sql.db_api_exceptions import ProgrammingError
+
+
+class Cursor:
+    def __init__(self):
+        self._results = None
+        self._row_index = 0
+        self.arraysize = 1
+        self._description_cache = None
+
+    def executemany(self, query):
+        self.execute(query)
+
+    def execute(self, query):
+        try:
+            self._results = exec(query=query, raw_results=True)
+            self._description_cache = None
+        except Exception as e:
+            self._results = None
+            raise DataError(f"Error executing query: {e}") from e
+
+    def fetchmany(self, size=-1):
+        if self._results is None or len(self._results) == 0:
+            return None
+
+        if size == -1:
+            size = self.arraysize
+
+        if size + self._row_index > len(self._results):
+            # give all that is remaining
+            size = len(self._results) - self._row_index
+
+        if size == 0:
+            # There are no more records
+            return None
+
+        rows = self._results[self._row_index : self._row_index + size]
+        self._row_index += size
+        return rows.to_pylist()
+
+    def nextset(self):
+        raise NotSupportedError("Operation not supported")
+
+    def setinputsizes(self, sizes):
+        raise NotSupportedError("Operation not supported")
+
+    def setoutputsize(self, sizes):
+        raise NotSupportedError("Operation not supported")
```

Comment:
I thought the user should now that the operation is not supported so I left it as it is. Let me know what you think.

Reply:
Since these are both essentially hints to the engine rather than operations that affect the semantics of querying, I think ignoring them is the right thing to do:

> Implementations are free to have this method do nothing and users are free to not use it.